### ISSUE TITLE
webContents.invoke() and ipcRenderer.handle()

### ISSUE THREAD

**Issue by jjeff** [electron/electron#25667]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description
There is no way of requesting data from the main process using promises in the same way that you can with `ipcRenderer.invoke()` and `ipcMain.handle()`

### Proposed Solution
Methods for `webContents.invoke()` and `ipcRenderer.handle()`

### Alternatives Considered
There are a few 3rd party libraries that can do this. [electron-promise-ipc](https://www.npmjs.com/package/electron-promise-ipc) is one. But since `ipcRenderer.invoke()` is native (with promise delivered from main to renderer), it makes sense that you'd be able to do the same thing in the other direction.

**Comment by nornagon** [comment]
This was a deliberate omission when adding `invoke()`, due to the fact that renderer processes cannot be trusted to return a response in a particular time frame, unlike the main process. It might make sense to add an API similar to `invoke` for the main->renderer direction, but an important consideration would be the ability to set a timeout after which, if no response has been received, the promise will be rejected. (Similarly, the promise should be rejected if the target frame is navigated, or if the process containing the target frame is terminated, or if for any other reason we are notified of any event which would imply that no response will ever be delivered.)

**Comment by jjeff** [comment]
Yeah. That's what I suspected. It would still be great functionality to have though!

**Comment by m4heshd** [comment]
I understand the reasoning but I still can't see why it shouldn't be there. Isn't it up to the developer to determine if there will be a response or not and handle the functionality accordingly? It doesn't make sense to write two IPC events in the main to do this at the moment.
